File: data-structure/c-cpp/c/Problem/Problem6-1/src/Stack.c
#include "Stack.h"
#include <stdlib.h>
#include <stdio.h>
void init_stack(Stack* stack) {
    stack -> plist = (List*)malloc(sizeof(List));
    init(stack -> plist);
}

void push(Stack* stack,Data data) {
    insert(stack -> plist,data);
}

Data pop(Stack* stack) {
    if (empty(stack)) {
        printf("stack is empty");
        exit(-1);
    }
    Data data;
    if (stack -> plist -> cur == NULL) {
        first(stack-> plist,&data);
    }else {
        next(stack->plist, &data);
    }
    return data;
}
Data peek(Stack* stack) {
    if (empty(stack)) {
        printf("stack is empty");
        exit(-1);
    }
    Data data;
    if (stack -> plist -> cur == NULL) {
        first(stack-> plist,&data);
        stack -> plist -> cur = NULL;
    }else {
        next(stack->plist, &data);
        stack -> plist -> cur = stack -> plist -> before;
    }
    return data;
}

int empty(Stack* stack) {
    return stack -> plist -> cur == stack -> plist -> tail;
}

File: data-structure/java/src/stack/array/Stack.java
package stack.array;

public class Stack <T>{
    private ArrayList<T> arrayList;

    Stack(){
        this.arrayList = new ArrayList<>();
    }
    public void push(T data){
        this.arrayList.insert(data);
    }
    public T peek(){
        return null;
    }
    public T pop(){
        return null;
    }
    public boolean empty(){
        return this.arrayList.count() == 0;
    }
}
